torch/csrc/jit/testing/file_check.cpp

```diff
@@ -293,6 +298,33 @@ struct FileCheckImpl {
     }
   }
 
+  // Checks that source token is highlighted, does not advance search range.
+  void doCheckSourceHighlighted(
+      const Check& check,
+      const std::shared_ptr<Source>& source,
+      size_t start_offset) {
+    auto pos = source->text().find(check.search_str_, start_offset);
+    if (pos == std::string::npos) {
+      // Guaranteed to fail to generate error message.
+      assertFind(source, check.search_str_, start_offset, check);
+      return;
+    }
+
+    auto lineno = source->lineno_for_offset(pos);
+    auto col = pos - source->offset_for_line(lineno);
+    auto highlight_lineno = lineno + 1;
+
+    auto highlight_start_offset =
+        source->offset_for_line(highlight_lineno) + col;
+    auto highlight_end_offset = std::min(
+        highlight_start_offset + check.search_str_.size(),
+        source->text().size());
+
+    SourceRange highlight_range(
+        source, highlight_start_offset, highlight_end_offset);
+    assertFind(highlight_range, std::string(highlight_range.size(), '~'));
```

Comment:
This checks that the given token is highlighted, but I think it is important to check that other parts of the same line are not. It seems to me that as written, this check would pass on a string like this:
```
do_not_highlight_me, highlight_me, do_not_highlight_me
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I think an approach that allows for some tolerance here (e.g. 5 extra ~ on either side) would be fine too. My goal here is to make sure that the highlight is useful and highlights only the intended portion of the source code.

What do you think?

Reply:
Good point. I added check to verify only requested token is highlighted. I am a bit on-the-fence about whether we should allow some tolerance though, which can be confusing and sometimes weakens the test. So I haven't added the tolerance yet. Let me know what you think.

Reply:
Good point